bundles/org.openhab.binding.ipcamera/README.md

```diff
@@ -0,0 +1,1312 @@
+To do this:
+
++ Consider setting the String channel that is called `mp4Filename` to a date and time stamp in a format that you like, or leave the channel empty for the filename to default to `ipcamera.mp4`.
++ Change the Number channel called `recordMp4` to a number of how many seconds that you wish to record for. The recording will then start.
++ Once the file is created the channel `recordMp4` will change itself back to 0 which can be used to trigger a rule to send the file, or you could use this event to change a counter variable that is used in the filename to create `visitor1.mp4 visitor2.mp4`.
++ You can use the ``FFMPEG_MP4_OUT_ARGUMENTS`` config to apply any FFmpeg filters to the output file.
++ The channel ``mp4History`` keeps a string of the last 50 recording filenames (separated by commas) until you reset the history. The channel mp4Filename is where this channel gets the names from when a recording is triggered.
++ The channel ``mp4HistoryLength`` keeps track of how many filenames are in the mp4History String. You can send the '0' command to this channel to clear the mp4History string at the same time as setting this channel back to 0.
+
+There is also a Habpanel Widget worth checking out as the thread has an example on how to use the bindings mp4 filename history feature to track the filenames of recent recordings.
+<https://community.openhab.org/t/custom-widget-camera-history-and-live-popup/103082>
+
+**NOTE:** If you are using a tmpfs folder, you will need to ensure you do not run out of space. I use a rule to move the files out of the tmpfs as they are created.
+
+
+*.items
+
+```java
+Switch BackyardCamExternalMotion "External Motion" { channel="ipcamera:HIKVISION:BackyardCam:externalMotion" }
+String BackyardCamMp4Filename "Backyard Mp4 Filename" { channel="ipcamera:HIKVISION:BackyardCam:mp4Filename" }
+Number BackyardCamRecordMp4 "Backyard seconds to Record" { channel="ipcamera:HIKVISION:BackyardCam:recordMp4" }
+
+```
+
+*.rules
+
+```java
+
+rule "Record MP4 on External Zwave PIR"
+    when
+    Item BackyardCamExternalMotion changed to ON
+    then
+        BackyardCamMp4Filename.sendCommand( new DateTimeType().toString ) //create a file with the timestamp as the filename.
+        BackyardCamRecordMp4.sendCommand(5) // record 5 seconds    
+end
+
+rule "Do something when recording finished"
+    when
+    Item BackyardCamRecordMp4 changed to 0
+    then
+    var TimeStamp = BackyardCamMp4Filename.state
+    logInfo("camera.rules", "Mp4 recording "+ TimeStamp.toString +".mp4 is ready to be used.")
+end
+
+```
+
+
+## MJPEG Streams
+
+Cameras that have built in MJPEG abilities can stream to openHAB with the MJPEG format with next to no CPU load, less than 1 second lag, and FFmpeg does not need to be installed.
+Cameras without this ability can still use this binding to convert their cameras RTSP h264 format to mjpeg (keep reading for more on this below) and this will take a lot of CPU power to handle the conversion. 
```

Comment:
```suggestion
Cameras without this ability can still use this binding to convert their RTSP h264 format to mjpeg (keep reading for more on this below) and this will take a lot of CPU power to handle the conversion. 
```

Reply:
Fixed.